arangosh/Restore/RestoreFeature.cpp

```diff
@@ -861,13 +861,16 @@ void RestoreFeature::start() {
 
     // re-check connection and version
     result =
-        _clientManager.getConnectedClient(httpClient, _options.force, true);
+        _clientManager.getConnectedClient(httpClient, _options.force, true, true);
     if (result.fail() &&
         !(_options.force && result.is(TRI_ERROR_INCOMPATIBLE_VERSION))) {
       LOG_TOPIC(FATAL, arangodb::Logger::FIXME)
           << "cannot create server connection, giving up!";
       FATAL_ERROR_EXIT();
     }
+  } else if (result.fail() && !_options.force) {
+    LOG_TOPIC(FATAL, arangodb::Logger::FIXME) << result.errorMessage();
+    FATAL_ERROR_EXIT();
   }
```

Comment:
Maybe combine the two checks since both are fatal, and just distinguish error messages?
```
if (result.fail() && !_options.force) {
  if (result.is(TRI_ERROR_INCOMPATIBLE_VERSION)) {
    LOG_TOPIC(FATAL, arangodb::Logger::FIXME)
        << "cannot create server connection, giving up!";
  } else {
    LOG_TOPIC(FATAL, arangodb::Logger::FIXME) << result.errorMessage();
  }
  FATAL_ERROR_EXIT();
}
```
It seems a little clearer to me this way at least...

Reply:
Changes implemented now.